include/kernel.h

```diff
@@ -5861,6 +5861,29 @@ extern void k_sys_runtime_stats_enable(void);
  */
 extern void k_sys_runtime_stats_disable(void);
 
+/**
+ * @brief Wait for an expression to return true with a timeout
+ *
+ * Spin on an expression with a timeout and optional delay between iterations
+ *
+ * Commonly needed when waiting on hardware to complete an asynchronous
+ * request to read/write/initialize/reset, but useful for any expression.
+ *
+ * @param expr Trueth expression you wish to spin on, ex: XYZREG & XYZREG_EN
+ * @param timeout Timeout to wait for, ex: K_MSEC(100)
```

Comment:
Timeout values are ticks internally.  It's possible that's going to be too coarse for this application on some platforms.  Seems like cycles might be a better fit.  Or not, I guess; timeouts do have the advantage of a clean API and robust conversion library.

Reply:
Maybe a compromise for ease of use, microseconds converted to cycles internally is what I landed on. Since ticks are so far apart I think you are right